core/src/test/java/io/substrait/relation/VirtualTableScanTest.java

```diff
@@ -0,0 +1,52 @@
+package io.substrait.relation;
+
+import static io.substrait.expression.ExpressionCreator.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+import io.substrait.expression.Expression;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class VirtualTableScanTest {
+
+  @Test
+  void check() {
+    VirtualTableScan virtualTableScan =
+        ImmutableVirtualTableScan.builder()
+            .addDfsNames(
+                "string",
+                "struct",
+                "struct_field1",
+                "struct_field2",
+                "list",
+                "list_struct_field1",
+                "map",
+                "map_key_struct_field1",
+                "map_value_struct_field1")
+            .addRows(
+                struct(
+                    false,
+                    string(false, "string_val"),
+                    struct(
+                        false,
+                        string(false, "struct_field1_val"),
+                        string(false, "struct_field2_val")),
+                    list(false, struct(false, string(false, "list_struct_field1_val"))),
+                    map(
+                        false,
+                        mapOf(
+                            struct(false, string(false, "map_key_struct_field1_val")),
+                            struct(false, string(false, "map_value_struct_field1_val"))))))
```

Comment:
If I understand how the naming stuff is supposed to work here, I believe only nested _structs_ contribute to names in the schema, so its not valid to have names for structs accessed _through_ lists or maps.

This is based on the example in https://substrait.io/tutorial/sql_to_substrait/#types-and-schemas

I'm not super familiar with how this is supposed to work though.

cc: @jacques-n

Reply:
Hmm, the example only has a list containing non-structs - so then there are no nested names. I don't see anything there indicating that a struct nested inside a list shouldn't contribute to names, and while it doesn't explicitly mention lists, I believe the part here would mean them as well: `Here, names is the names of all fields. In nested schemas, this includes the names of subfields in depth-first order.` It seems to me to make sense to include them, given SQL does allow names within structs inside lists as well :)

Isthmus does include nested fields from inside arrays:
```
./isthmus-cli/build/graal/isthmus "select col from t" -c "create table t (col row(sf1 int null, sf2 varchar) array)"
..
             "baseSchema": {
                "names": ["COL", "SF1", "SF2"],
                "struct": {
                  "types": [{
                    "list": {
                      "type": {
                        "struct": {
                          "types": [{
                            "i32": {
                              "typeVariationReference": 0,
                              "nullability": "NULLABILITY_NULLABLE"
                            }
                          }, {
..
```

Reply:
Yeah, structs insides lists can have names when output. Yes, the names should be included in the depth-first naming list.